test/cmd/newapp.sh

```diff
@@ -167,4 +167,7 @@ os::cmd::expect_success 'oc new-app mongo -o json | python -m json.tool'
 # Ensure custom branch/ref works
 os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world#beta4'
 
+# Ensure the resulting BuildConfig doesn't have unexpected sources
+os::cmd::expect_success_and_not_text 'oc new-app https://github.com/openshift/ruby-hello-world -o=jsonpath="{.items[?(@.kind==\"BuildConfig\")].spec.source}"' 'dockerfile|binary'
```

Comment:
> where?

I meant I added the `--output-version=v1` @liggitt asked for. Just saw your comment later, and answered in the previous comment above. Reiterating, I'm confident with the tests we have for the behavior change we introduced in #6615.